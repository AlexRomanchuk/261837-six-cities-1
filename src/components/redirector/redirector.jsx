import React from "react";
import {Redirect} from "react-router-dom";
import PropTypes from "prop-types";
import {connect} from "react-redux";

const Redirector = (props) => {
  const {offerId} = props;
  return <Redirect to={`/offer/${offerId}`} />;
};

Redirector.propTypes = {
  offerId: PropTypes.string.isRequired,
};

const mapStateToProps = ({}, ownProps) => Object.assign({}, ownProps, {
  offerId: ownProps.match.params.id
});

export {Redirector};

export default connect(mapStateToProps)(Redirector);
