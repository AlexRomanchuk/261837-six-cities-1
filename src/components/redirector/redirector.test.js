import React from "react";
import renderer from "react-test-renderer";
import {Redirector} from "../redirector/redirector.jsx";
import {StaticRouter} from "react-router-dom";

it(`correct renders redirector component`, () => {
  const tree = renderer.create(<StaticRouter>
    <Redirector
      offerId={`2`}
    /></StaticRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
