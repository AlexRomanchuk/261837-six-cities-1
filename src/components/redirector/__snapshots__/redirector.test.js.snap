// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct renders redirector component 1`] = `null`;
